### ISSUE TITLE
Support Scijava's @Parameter annotation

### ISSUE THREAD

**Issue by stelfrich** [imagej/imagej-ops#471]
It would be great to be able to use ops with the `@Parameter` annotation and as script parameter, as pointed out in a [forum thread](http://forum.imagej.net/t/script-parameters/3362/3?u=stelfrich).

Something like:
```groovy
// @Threshold(label="What Threshold to use?") threshold
// @Img input
// @OUTPUT Img output
output = threshold.compute(img)
```

I guess this requires establishing a connection between the requested op and input to enable matching?

**Comment by dietzc** [comment]
I'm not sure if I like this idea. Will we introduce special annotations for each and every type of input parameter? I think the idea with such plugins as  `Threshold` was that there is something like a  `Threshold` class which can be annotated. SciJava detects, that this actually a plugin of many types and offers you a drop-down to select the threshold. Basically the same idea, just on `class` or `plugin` level. Correct me if I'm wrong @ctrueden.

**Comment by stelfrich** [comment]
> Will we introduce special annotations for each and every type of input parameter?

`@Threshold` is an annotation just like `@Img`, isn't it? In the example, I am trying to request something like `Ops.Threshold.*`.

> I think the idea with such plugins as `Threshold`

About which plugin(s) are you talking? The deprecated [`ThresholdMethod`](https://github.com/imagej/imagej-deprecated/blob/master/src/main/java/net/imagej/threshold/ThresholdMethod.java) et al?

**Comment by dietzc** [comment]
Yes, I meant something like `ThresholdMethod` et al. If we introduce them, I think there are mechanisms in SciJava that `@Threshold` in the script world would deliver the plugins as a drop-down. I think the pattern is https://github.com/scijava/scijava-common/blob/master/src/main/java/org/scijava/plugin/SingletonService.java.

**Comment by stelfrich** [comment]
> I think there are mechanisms in SciJava that `@Threshold` in the script world would deliver the plugins as a drop-down.

That's right. The following Groovy script produces the mentioned drop-down:
```groovy
// @net.imagej.threshold.ThresholdMethod threshold
```

I was just thinking out loud about adding something like that on the Ops level (since `ThresholdMethod` el al are deprecated) as well. AFAICT, implementations of `ThresholdMethod` are `Plugin`s themselves, that's why Scijava picks them up.

**Comment by ctrueden** [comment]
I definitely have thoughts, opinions and plans surrounding this, and have for a long time. I will try to elaborate in this here issue when I get a chance, but there will be some delay.
